### PR TITLE
#160056368: setup test coverage

### PR DESCRIPTION
**Pull request by chidioguejiofor**
#### What does this PR do?
- installs jest and enzyme
- added jest.config to the app
- updates CircleCI config to run tests
- adds project to Code-climate
- ensures that build is passing on Circleci
- separates script for test and coverage
- adds repo token environmental variable to CircleCI
-adds badges to the README

#### Description of Task to be completed?
- acquire test coverage badge from code-climate
- configure test environment
- configure CircleCI to run test

#### How should this be manually tested?
- run `git pull` from console
- run `npm run test` to see test result
- run `npm run coverage` to see coverage results
- go to circleCi to view  branch build via https://circleci.com/gh/andela/rivendell-ah-client/127

#### Any background context you want to provide?
- None

#### What are the relevant pivotal tracker stories?
- 160056368 
- 160056367 
- 160056371

#### Screenshots (if appropriate)
screen shot 2018-09-06 at 1 23 49 am

#### Questions:
- None

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    enabled: false
+    config:
+      languages:
+      - javascript
+exclude_patterns:
+  - tests/**/*
+  - public/**/*
+  - webppack.config.js
+  - jest.config.js
+
```
Comment (mppdg) [inline .codeclimate.yml]: EOL

File: .eslintignore [hunk 2]
```diff
@@ -1 +1,2 @@
 webpack.config.js
+tests/**/*.test.js
```
Comment (mppdg) [inline .eslintignore]: EOL

File: .codeclimate.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+plugins:
+  duplication:
+    enabled: false
+    config:
+      languages:
+      - javascript
+exclude_patterns:
+  - tests/**/*
+  - public/**/*
+  - webppack.config.js
```
Comment (mppdg) [inline .codeclimate.yml]: `webppack.config.js` wrongly spelt. Remove one of the Ps

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  setupFiles: [
+    '<rootDir>/tests/config/setup.js',
+  ],
+  collectCoverageFrom: ['<rootDir>/src/**/*.{js,jsx}'],
+  coveragePathIgnorePatterns: [
```
Comment (rhenshaw56) [inline jest.config.js:7]: add an option for a testUrl, so we don't have to come back for that

File: jest.config.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    '<rootDir>/tests',
+    '<rootDir>/src/index.js',
+  ],
+  snapshotSerializers: ['enzyme-to-json/serializer'],
+  moduleNameMapper: {
+    '\\.(css|scss)$': 'identity-obj-proxy',
```
Comment (rhenshaw56) [inline jest.config.js]: you would also need to add for png, eof, woff types also

File: tests/config/setup.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import Enzyme from 'enzyme';
```
Comment (rhenshaw56) [inline tests/config/setup.js]: change to enzyme

**Review by rhenshaw56 — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 7]
```diff
@@ -1,3 +1,4 @@
+coverage
 node_modules
 <<<<<<< HEAD
```
Comment (JCanaks) [inline .gitignore]: remove conflict head

Comment (JCanaks) [inline .gitignore]: resolve conflict

**Review by JCanaks — APPROVED** [review]

**Review by mppdg — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]

**Pull request merged**